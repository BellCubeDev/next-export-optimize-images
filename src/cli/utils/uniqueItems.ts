import uniqWith from 'lodash.uniqwith'

import type { Manifest } from '../'

const uniqueItems = (items: Manifest) => uniqWith(items, (a, b) => a.output === b.output)

export default uniqueItems
