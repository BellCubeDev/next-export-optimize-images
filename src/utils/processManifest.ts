const processManifest = (manifestJson: string) =>
  manifestJson
    .trim()
    .split(/\n/g)
    .map((line) => JSON.parse(line))

export default processManifest
