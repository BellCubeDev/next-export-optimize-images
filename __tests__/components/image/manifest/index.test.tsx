/**
 * @jest-environment jsdom
 */
process.env['TEST_JSON_PATH'] = '__tests__/components/image/manifest/manifest.json'

import path from 'path'

import { render } from '@testing-library/react'
import fs from 'fs-extra'
import React from 'react'

import uniqueItems from '../../../../src/cli/utils/uniqueItems'
import CustomImage from '../../../../src/image'
import processManifest from '../../../../src/utils/processManifest'

const manifestPath = path.resolve(__dirname, 'manifest.json')

beforeAll(async () => {
  await fs.remove(manifestPath)
})

describe('Create JSON', () => {
  test('Common', () => {
    render(
      <>
        <CustomImage src="/img.png" width={1920} height={1280} priority alt="" />
      </>
    )

    const manifest = uniqueItems(processManifest(fs.readFileSync(manifestPath, 'utf-8')))
    expect(manifest).toMatchSnapshot()
  })
})
