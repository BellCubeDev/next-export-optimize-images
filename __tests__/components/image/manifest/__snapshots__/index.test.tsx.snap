// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create JSON Common 1`] = `
[
  {
    "extension": "png",
    "output": "/_next/static/chunks/images/img_1920_75.png",
    "quality": 75,
    "src": "/img.png",
    "width": 1920,
  },
  {
    "extension": "png",
    "output": "/_next/static/chunks/images/img_3840_75.png",
    "quality": 75,
    "src": "/img.png",
    "width": 3840,
  },
]
`;
